internal/subscriber/subscriber.go

```diff
@@ -88,6 +89,8 @@ func (s *Subscriber) GetSubscription() string {
 
 // commits existing message on primary topic and pushes message to the pre-defined retry topic
 func (s *Subscriber) retry(ctx context.Context, retryMsg *RetryMessage) {
+	retryMsg.incrementRetry()
```

Comment:
should this be incremented here or after the check https://github.com/razorpay/metro/pull/89/files#diff-4bcad8e79804b094286e6698fe58e6f08d402b61a217a004fd4b078034ee9992R107?

Reply:
changed